nixos/modules/services/network-filesystems/samba.nix

```
@@ -5,7 +5,9 @@ with lib;
 let
   cfg = config.services.samba;
 
-  settingsFormat = pkgs.formats.ini { };
+  settingsFormat = pkgs.formats.ini {
+    listToValue = lib.concatMapStringsSep " " (generators.mkValueStringDefault { });
```

This could be
```suggestion
    listToValue = lib.concatStringsSep " ";
```
using https://github.com/NixOS/nixpkgs/pull/341248